From 670b2d27b82af8091e462d8cfc1edeedbbaed6e0 Mon Sep 17 00:00:00 2001
From: Jeferson Oliveira <jroliveira.oliveira301@gmail.com>
Date: Wed, 16 Aug 2023 19:36:59 +0200
Subject: [PATCH] Restrict apps access to /proc/net/unix

* Banking apps read the file /proc/net/unix to
  check system modifications, (in our case, custom rom)
  Let's avoid this by chmod it 440.

Test: Install a banking app as of Brazilian Nubank or PicPay, login.
      Observed no side effects on any other proccess or apps.
Change-Id: Ic5f91e195b0b3b60fffff4eb0c427b6ed5574a67
Signed-off-by: Majaahh <maja.blaszczak@proton.me>
---
 init.s5e8825.rc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/init.s5e8825.rc b/init.s5e8825.rc
index 6d0f709..2edc34e 100644
--- a/init.s5e8825.rc
+++ b/init.s5e8825.rc
@@ -964,6 +964,9 @@ on property:sys.boot_completed=1
     write /dev/cpuset/background/cpus 0-1
     write /dev/cpuset/sf/cpus 6-7
 
+    # Only owner can acces this node
+    chmod 440 /proc/net/unix
+
 # KERNEL CORE MEMORY
 on boot
     write /proc/sys/vm/swappiness 130
-- 
2.52.0

